pulldown-cmark/src/parse.rs

```diff
@@ -831,10 +878,117 @@ impl<'input, F: BrokenLinkCallback<'input>> Parser<'input, F> {
+    /// acts as a control flow guard. Returns the link node if a wikilink was
+    /// found and created.
+    fn handle_wikilink(
+        &mut self,
+        block_text: &'input str,
+        cur_ix: TreeIndex,
+        prev: Option<TreeIndex>,
+    ) -> Option<TreeIndex> {
+        let next_ix = self.tree[cur_ix].next.unwrap();
+        // this is a wikilink closing delim, try popping from
+        // the wikilink stack
+        if let Some(tos) = self.wikilink_stack.pop() {
+            if tos.ty == LinkStackTy::Disabled {
+                return None;
+            }
+            // fetches the beginning of the wikilink body
+            let Some(body_node) = self.tree[tos.node].next.and_then(|ix| self.tree[ix].next) else {
+                // skip if no next node exists, like at end of input
+                return None;
+            };
+            let start_ix = self.tree[body_node].item.start;
+            let end_ix = self.tree[cur_ix].item.start;
+            let wikilink = match scan_wikilink_pipe(
+                block_text,
+                start_ix, // bounded by closing tag
+                end_ix - start_ix,
+            ) {
+                Some((rest, wikiname)) => {
+                    // bail early if the wikiname would be empty
+                    if wikiname.is_empty() {
+                        return None;
+                    }
+                    // [[WikiName|rest]]
+                    let body_node = scan_nodes_to_ix(&self.tree, Some(body_node), rest);
+                    if let Some(body_node) = body_node {
+                        // break node so passes can actually format
+                        // the display text
+                        self.tree[body_node].item.start = start_ix + rest;
+                        Some((body_node, wikiname))
+                    } else {
+                        None
+                    }
+                }
+                None => {
+                    // [[WikiName]]
+                    let wikiname = &block_text[start_ix..end_ix];
+                    // or [[Nested/WikiName]]
+                    let display_ix = wikiname
+                        .as_bytes()
+                        .iter()
+                        .rposition(|b| *b == b'/')
+                        .map(|ix| ix + 1)
+                        .unwrap_or(0)
+                        + start_ix;
+                    let display_end_ix = wikiname
+                        .as_bytes()
+                        .iter()
+                        .position(|b| *b == b'#')
+                        .unwrap_or(wikiname.len())
+                        + start_ix;
+                    // bail early if the wikiname would be empty
+                    if display_ix >= display_end_ix {
+                        return None;
+                    }
+                    // TODO: wikitext should not be styled, might
+                    // need a more experienced contributor's help
```

Comment:
Can you give an example of where the wrong thing happens? Because it seems like you *have* ignored styling. In the other branch, you seem to have found the existing tree node after the pipe (probably a text node), and made it the child of the wikilink. If there's a MaybeEmphasis in there, it will be retained. In this branch, however, you've completely replaced the children of this tree node with plain text.

As a result, this test case passes on the PR as it stands:

~~~
```````````````````````````````` example_wikilinks
[[**not**]]

[[yes|**yes**]]
.
<p><a href="**not**">**not**</a></p>
<p><a href="yes"><strong>yes</strong></a></p>
````````````````````````````````
~~~